ICSharpCode.Decompiler/DebugInfo/IDebugInfoProvider.cs

```diff
@@ -23,6 +23,7 @@ public interface IDebugInfoProvider
 		IList<SequencePoint> GetSequencePoints(MethodDefinitionHandle method);
 		IList<Variable> GetVariables(MethodDefinitionHandle method);
 		bool TryGetName(MethodDefinitionHandle method, int index, out string name);
+		bool TryGetExtraTypeInfo(MethodDefinitionHandle method, int index, out string[] tupleElementNames, out bool[] dynamicFlags);
```

Comment:
In general I am fine with out parameters, but it might be beneficial to use a class/struct instead to avoid binary breaking changes in interface methods. So if there is a new language feature we want to add support for, we don't have to add extra overloads or do a new major version. What do you think?

Reply:
I'd agree with introducing a struct for the out parameters to make way for future information that could find its way into PDB files. 

I think something like this would be fine:
```csharp
bool TryGetExtraTypeInfo(MethodDefinitionHandle method, int index, out PdbExtraTypeInfo extraTypeInfo);
```

```csharp
struct PdbExtraTypeInfo {
	public string[] TupleElementNames;
	public bool[] DynamicFlags;
}
```

Reply:
Yeah, I think that's fine.